pymatgen/io/vasp/sets/base.py

```diff
@@ -0,0 +1,1191 @@
+        include_cif: bool = False,
+        potcar_spec: bool = False,
+        zip_output: bool = False,
+    ) -> None:
+        """
+        Writes a set of VASP input to a directory.
+
+        Args:
+            output_dir (str): Directory to output the VASP input files
+            make_dir_if_not_present (bool): Set to True if you want the
+                directory (and the whole path) to be created if it is not
+                present.
+            include_cif (bool): Whether to write a CIF file in the output
+                directory for easier opening by VESTA.
+            potcar_spec (bool): Instead of writing the POTCAR, write a "POTCAR.spec".
+                This is intended to help sharing an input set with people who might
+                not have a license to specific Potcar files. Given a "POTCAR.spec",
+                the specific POTCAR file can be re-generated using pymatgen with the
+                "generate_potcar" function in the pymatgen CLI.
+            zip_output (bool): If True, output will be zipped into a file with the
+                same name as the InputSet (e.g., MPStaticSet.zip)
+        """
+        if potcar_spec:
+            vasp_input = None
+            if make_dir_if_not_present:
+                os.makedirs(output_dir, exist_ok=True)
+
+            with zopen(f"{output_dir}/POTCAR.spec", mode="wt") as file:
+                file.write("\n".join(self.potcar_symbols))
+
+            for key in ["INCAR", "POSCAR", "KPOINTS"]:
+                if (val := getattr(self, key.lower())) is not None:
+                    with zopen(os.path.join(output_dir, key), mode="wt") as file:
+                        file.write(str(val))
+        else:
+            vasp_input = self.get_input_set()
+            vasp_input.write_input(output_dir, make_dir_if_not_present=make_dir_if_not_present)
+
+        cif_name = ""
+        if include_cif and vasp_input is not None:
+            struct = vasp_input["POSCAR"].structure
+            cif_name = f"{output_dir}/{struct.formula.replace(' ', '')}.cif"
+            struct.to(filename=cif_name)
+
+        if zip_output:
+            filename = f"{type(self).__name__}.zip"
+            with ZipFile(os.path.join(output_dir, filename), mode="w") as zip_file:
+                for file in ["INCAR", "POSCAR", "KPOINTS", "POTCAR", "POTCAR.spec", cif_name]:
+                    try:
+                        zip_file.write(os.path.join(output_dir, file), arcname=file)
+                    except FileNotFoundError:
+                        pass
+
+                    try:
+                        os.remove(os.path.join(output_dir, file))
+                    except (FileNotFoundError, PermissionError, IsADirectoryError):
+                        pass
+
+        for key, val in self.files_to_transfer.items():
+            with zopen(val, "rb") as fin, zopen(str(Path(output_dir) / key), "wb") as fout:
+                shutil.copyfileobj(fin, fout)
```

Comment:
Strictly speaking, `write_input` should be attached to the `InputSet` rather than the generator (although I see that we're conflating the two here for backward compatibility reasons). That's OK, but is there a way to 

1) make the call signature more compatible with the base class (e.g., `directory` instead of `output_dir`) and

2) Actually use the underlying `write_input` method of `self.get_input_set` instead of relying on a custom implementation here?

One of my main objectives in implementing the abc was to achieve a (more) consistent file IO call signature across different `InputSet`.

That's my only suggestion - I'll defer to @janosh and others on the thread if it's feasible to implement or not.

Reply:
Thanks, @rkingsbury. I've moved the features of `VaspInputSet.write_input` to `VaspInput.write_input` (`VaspInput` is the underlying class), and made `VaspInputSet.write_input` an interface to the underlying method

Re the `output_dir` --> `directory` stylistic change: seems like the VASP classes use `output_dir` rather than `directory`. While it's better to have a cohesive style across PMG, I'd like to avoid breaking changes with this PR. I can add a `directory` kwarg that aliases to `output_dir` and specify a deprecation period for it if that's preferable

Reply:
Keep output_dir. "Directory" is unclear. While the context would suggest an output, there are instances where there is both an input_dir and an output_dir. It also goes too close to the `dir` special function in python.